include/rdma/fi_eq.h

```diff
@@ -237,7 +242,10 @@ struct fi_cq_err_entry {
 	int			err;
 	int			prov_errno;
 	/* err_data is available until the next time the CQ is read */
-	void			*err_data;
+	union {
+		void				 *err_data;
+		struct fi_cq_err_source_err_data *src_err_data;
+	};
```

Comment:
Yes. I need to fix this. I'll align better to current definition of err_data and err_data_size.

The following is what I was trying to accomplish. To support FI_SOURCE_ERR with these auth key AV extensions, both addr and auth_key need to be returned. Since the majority of the CQ error fields may be valid with FI_SOURCE_ERR, the user needs to provide an structure (e.g. fi_cq_err_source_err_data) where the provider can fill in the addr and auth_key fields.

Reply:
Here is what I am thinking.

```
struct fi_cq_err_entry {
	void			*op_context;
	uint64_t		flags;
	size_t			len;
	void			*buf;
	uint64_t		data;
	uint64_t		tag;
	size_t			olen;
	int			err;
	int			prov_errno;
	/* err_data is available until the next time the CQ is read */
	void			*err_data;
	size_t			err_data_size;
	void			*auth_key;
	size_t			auth_key_size;
};

For EPs configured with FI_SOURCE_ERR and bound to an AV with
FI_AUTH_KEY, when FI_EADDRNOTAVAIL CQ errors occur, users need to
provide a buffer via the fi_cq_err_entry::auth_key and
fi_cq_err_entry::auth_key_size fields to retrieve the auth_key for the
event. fi_cq_err_entry::err_data and fi_cq_err_entry:err_data_size will
be used, as normal, to return addr. The returned addr and auth_key can
be used for fi_av_insert_auth_key_addr().
```